quickwit-cli/src/split.rs

```diff
@@ -129,12 +201,100 @@ impl SplitCliCommand {
 
     pub async fn execute(self) -> anyhow::Result<()> {
         match self {
+            Self::ListSplit(args) => list_split_cli(args).await,
             Self::DescribeSplit(args) => describe_split_cli(args).await,
             Self::ExtractSplit(args) => extract_split_cli(args).await,
         }
     }
 }
 
+pub async fn list_split_cli(args: ListSplitArgs) -> anyhow::Result<()> {
+    debug!(args = ?args, "list-split");
+
+    let metastore_uri_resolver = MetastoreUriResolver::default();
+    let metastore = metastore_uri_resolver.resolve(&args.metastore_uri).await?;
+    let time_range_opt = match (args.from, args.to) {
+        (None, None) => None,
+        (None, Some(to)) => Some(Range {
+            start: i64::MIN,
+            end: to,
+        }),
+        (Some(from), None) => Some(Range {
+            start: from,
+            end: i64::MAX,
+        }),
+        (Some(from), Some(to)) => Some(Range {
+            start: from,
+            end: to,
+        }),
+    };
+    let is_disjoint_time_range = |left: &Range<i64>, right: &RangeInclusive<i64>| {
+        left.end <= *right.start() || *right.end() < left.start
+    };
+    let filter_tag_set = args.tags.iter().cloned().collect::<BTreeSet<_>>();
+
+    let mut splits = vec![];
+    // apply tags & time range filter.
+    for split in metastore.list_all_splits(&args.index_id).await? {
+        if !filter_tag_set.is_empty() && split.split_metadata.tags.is_disjoint(&filter_tag_set) {
```

Comment:
I don't think the rule is great here. because it does not include the tag:* gig.

That being said,I suggested merge this PR as is and open an issue to address that later, because of the discussion in #887.